### ISSUE TITLE
Fix S3005 FN: Support generic attributes

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6396]
### Description

FN for S3005 (ThreadStaticNonStaticField).

Using C#11 generic attributes, the rule is not raising an issue.

### Repro steps

```csharp
    public class ThreadStaticNonStaticField
    {
        [ThreadStaticAttribute<int>]  // FN
        private int count1 = 0, count11 = 0;
    }

    public class ThreadStaticAttribute<T> : ThreadStaticAttribute { }

```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by csaba-sagi-sonarsource** [comment]
The rule actually does not support derived attributes at all, and it will not be fixed as it is an edge case and would have a heavy performance impact on the rule. The decision was documented in https://github.com/SonarSource/sonar-dotnet/pull/6453
